### PR TITLE
Add proxy_byos_mode column

### PR DESCRIPTION
**Pull request by jesusbv**
## Description

Before LTSS support proxy_byos was a boolean column, representing PAYG or BYOS

With LTSS, we need a column (proxy_byos_mode) to represent one more value, change the type to enum with the following mapping:

PAYG -> 0
BYOS -> 1
LTSS -> 2

## How to test 

Create a system with `:hybrid`, should not produce an error
Check that the just created system has `hybrid` to true

## Change Type

*Please select the correct option.*

- [ ] **Bug Fix** (a non-breaking change which fixes an issue)
- [X] **New Feature** (a non-breaking change which adds new functionality)
- [ ] **Documentation Update** (a change which only updates documentation)

## Checklist

*Please check off each item if the requirement is met.*

- [X] I have reviewed my own code and believe that it's ready for an external review.
- [ ] I have provided comments for any hard-to-understand code.
- [ ] I have documented the `MANUAL.md` file with any changes to the user experience.
- [ ] If my changes are non-trivial, I have added a changelog entry to notify users at `package/obs/rmt-server.changes`.

## Review

Please check out our [review guidelines](https://github.com/SUSE/scc-docs/blob/master/team/workflow/code_review.md) 
and get in touch with the author to get a shared understanding of the change.

### CODE REVIEW

**Comment by jesusbv** [comment]
Tests fail, but I think the spec files should be adapted on a different PR

that way we have one PR/commit for the db migration alone

**Comment by jesusbv** [comment]
We could also change the name from `proxy_byos` to `proxy_byos_hybrid`, thoughts ?

File: db/migrate/20240729103525_update_proxy_byos_column_type.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class UpdateProxyByosColumnType < ActiveRecord::Migration[6.1]
+  def up
+    add_column :systems, :proxy_byos_mode, :integer, default: 0
+    safety_assured do
```
Comment (felixsch) [inline db/migrate/20240729103525_update_proxy_byos_column_type.rb:4]: will this work with millions of systems? Or is batched more safe here?

Comment (jesusbv) [reply db/migrate/20240729103525_update_proxy_byos_column_type.rb]: good point, I added [838d16a](https://github.com/SUSE/rmt/pull/1193/commits/838d16a2cc1de4797adbcf5f1ef9bedbd16947a0) for the batches approach

Comment (jesusbv) [reply db/migrate/20240729103525_update_proxy_byos_column_type.rb]: good point, [0b62c2b](https://github.com/SUSE/rmt/pull/1193/commits/0b62c2b1820767bb2e15543a0b6885699ea3f1ef) addresses the batch approach

**Review by rjschwei — APPROVED** [review]

**Review by digitaltom — APPROVED** [review]

**Review by felixsch — APPROVED** [review]

**Pull request merged**